src/test/steps/generateReportSteps.ts

```diff
@@ -0,0 +1,36 @@
+import { When, Then } from '@cucumber/cucumber';
+import { expect } from '@playwright/test';
+import basePage from '../../pages/basePage';
+
+When('User navigates to {string} from {string}', async function (tab, page) {
+  const tabName = await basePage.wrapper.toCamelCase(tab);
+  const pageName = await basePage.wrapper.toCamelCase(page);
+  await basePage[pageName][tabName].click();
+});
+
+When('User clicks {string} on {string}', async function (button, page) {
+  const buttonName = await basePage.wrapper.toCamelCase(button);
+  const pageName = await basePage.wrapper.toCamelCase(page);
+  await basePage.wrapper.waitAndClick(basePage[pageName][buttonName]);
+});
+
+When('User chooses {string} for {string} on {string}', async function (option, dropdown, page) {
+  const optionName = await basePage.wrapper.toCamelCase(option);
+  const dropdownName = await basePage.wrapper.toCamelCase(dropdown);
+  const pageName = await basePage.wrapper.toCamelCase(page);
```

Comment:
You've repetitively used `toCamelCase` method. You can refactor and simplify it. For example:
```
const [ casedArgument1, casedArgument2, casedArgument3 ] = toCamelCase(argument1, argument2, argument)
```
You can change `toCamelCase` to accept an array of parameters, transform them into a camel case, and then return an array of transformed arguments.